x/migration/keeper/msg_server_claim_morse_supplier.go

```diff
@@ -29,8 +29,55 @@ func (k msgServer) ClaimMorseSupplier(
 ) (*migrationtypes.MsgClaimMorseSupplierResponse, error) {
 	sdkCtx := cosmostypes.UnwrapSDKContext(ctx)
 	logger := k.Logger().With("method", "ClaimMorseSupplier")
+	waiveMorseClaimGasFees := k.GetParams(sdkCtx).WaiveMorseClaimGasFees
 
-	if err := msg.ValidateBasic(); err != nil {
+	// Ensure that gas fees are NOT waived if one of the following is true:
+	// - The claim is invalid
+	// - Morse account has already been claimed
+	// Claiming gas fees in the cases above ensures that we prevent spamming.
+	//
+	// Rationale:
```

Comment:
Can you think of a way of create a helper function so we have this comment only exist in one place?